utils/configuration_test.go

```diff
@@ -560,3 +560,18 @@ func TestSetSignalTrap(t *testing.T) {
 		require.NotNil(t, signalsPassedOn[s.String()])
 	}
 }
+
+// TODO: undo this extra indirection, needed for mocking notary.NotarySupportedSignals being empty, when we have
+// a windows CI system running
+func TestSetSignalTrap(t *testing.T) {
+	testSetSignalTrap(t)
+}
+
+func TestSetSignalTrapMockWindows(t *testing.T) {
+	old := notary.NotarySupportedSignals
+	notary.NotarySupportedSignals = nil
+	defer func() {
+		notary.NotarySupportedSignals = old
```

Comment:
I think not deferring this and moving it to after `testSetSignalTrap` will solve the race checker problem.